src/main/asciidoc/_chapters/cp.adoc

```diff
@@ -193,6 +209,12 @@ link:https://issues.apache.org/jira/browse/HBASE-5448[HBASE-5448]). To upgrade y
 HBase cluster from 0.94 or earlier to 0.96 or later, you need to reimplement your
 coprocessor.
 
+In HBase 2.0.0, we make use of a shaded version of protobuf 3.x, but still keep the
+protobuf for coprocessor on 2.5.0. In 3.0.0, we removed all dependencies on non-shaded
+protobuf so you need to reimplement your coprocessor to make use of the shaded protobuf
+version provided in hbase-thirdparty. Please see the <<protobuf,protobuf>> section for
```

Comment:
Oh, so you put together your own .pb file/pb Service, and then use the protobuf version over in hbase-thirdparty to generate stubs.

if you need to convert your pb to pojos, you build your own. If you want to use internal hbase pb or pojo types, I suppose you just use them?  They are not audience private but limitedprivate to CPEP implementors?

Reply:
I think this is why we need a hbase-protocol modue in the past? The CPEP implementation could reference the protobuf defination. And in general, adding breaking changes to proto files is a no no, it will break wire compatible, so I think it is fine to reference them?

Reply:
I was asking what the audience for protos is. If used by CPEP, then they are not Private. LimitedPrivate for CPEPs?

Reply:
Yes, I think it is LimitedPrivate for CPEPs. What I said above is that, we had this consensus in the past, otherwise we did not need to provide a hbase-protocol module which duplicates a lot of proto files...

Reply:
We should label it explicitly w/ the LimitedPrivate for CPEPs and in refguides. Another issue.